src/interop/js-interop/package-web.md

```diff
@@ -0,0 +1,195 @@
+**you need to migrate to `package:web` as soon as possible**.
+:::
+
+## Migrating from `dart:html`
+
+Remove the `dart:html` import and replace it with `package:web/web.dart`:
+
+```dart
+import 'dart:html' as html; // Remove
+import 'package:web/web.dart' as web; // Add
+```
+
+Add `web` to the `dependencies` in your pubspec:
+
+```yaml
+dependencies:
+  web: ^0.4.2
+```
+
+The following sections cover the current available methods of migration
+from `dart:html` to `package:web`.
+
+For any other migration issues, check the [dart-lang/web][] repo and file an issue.
+
+### Renames
+
+Many of the symbols in `dart:html` were renamed from
+their original IDL declaration to align more with Dart style.
+For example, `appendChild` became `append`,
+`HTMLElement` became `HtmlElement`, etc.
+
+In contrast, to be as lean and maintainable as possible,
+`package:web` is fully aligned with the IDL declaration
+and keeps all the original names.
+A `dart fix` is available to convert symbols that have been renamed
+between `dart:html` and `package:web`. 
+
+After changing the import, any renamed objects will be new "undefined" errors.
+You can adress these either:
+- From the CLI by running `dart fix --dry-run`.
+- In your IDE, select the `dart fix`: **Rename to '`package:web name`'**.
+
+{% comment %}
+Maybe a pic here of menu selection in IDE?
+{% endcomment -%}
+
+The `dart fix` should cover the majority of renames.
+If you come across a `dart:html` symbol without a built-in fix,
+first let us know by filing an [issue][].
+Then, you can manually discover the `package:web` name
+by looking up its `@Native` annotation.
+You can do this by either:
+
+- Ctrl or cmd clicking the name in the IDE and choosing **Go to Definition**.
+- Searching for the name in the [`dart:html` API docs][html] 
+  and checking its page under *Annotations*.
+
+### Type checks
+
+It's common for `dart:html` code to utilize runtime `is` checks
+to confirm whether an object actually maps to its JS type.
+In constrast, the types that represent browser bindings in `package:web`
+are not reified. This means a runtime `is` check can't confirm the interop type,
+only that the underlying type is JS `Object`.
+
+To be able to perform type checks, migrate any `dart:html` code
+using `is` type tests to use [interop methods][] like `instanceOf`.
+The [Compatibility, type checks, and casts][]
+section of the JS types page covers static interop alternatives to `is`
+in detail.
+
+```dart
+obj is Window; // Remove
+obj.instanceOf('Window'); // Add
+```
+
+### Method conversions
+
+To migrate unsupported method calls from `dart:html` to `package:web`,
+you'll mostly want to use conversion methods like `toJS()` and `toDart()`.
+Learn how to use static interop conversion methods from the [Conversions][]
+section of the JS types page.
+
+```dart
+_clickSubscription = window.onClick.listen(callback); // Remove
+window.onclick = callback.toJS; // Add
+```
```

Comment:
This example is from a doc written before the helper methods became a thing. I'm guessing technically there's a helper to address this kind of thing now, so maybe a bad example?

Reply:
There is, although the helpers are on `Element`, but yeah, I think the example is probably not what we want. We can put a TODO here to look at some of the migrations and look for a better example.